sklearn/feature_extraction/_hashing.pyx

```diff
@@ -43,7 +45,14 @@ def transform(raw_X, Py_ssize_t n_features, dtype):
 
     for x in raw_X:
         for f, v in x:
-            value = v
+            if isinstance(v, integer_types):
+                value = v
+            elif isinstance(v, string_types):
+                f = "%s%s%s" % (f, '=', v)
+                value = 1
+            elif isinstance(v, unicode):
```

Comment:
I don't understand when this case happens. Surely the `isinstance(v, string_types)` will already have been executed.